DXMainClient/DXGUI/Multiplayer/MapTextureCacheManager.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+using DTAClient.Domain.Multiplayer;
+
+using SixLabors.ImageSharp;
+
+namespace DTAClient.DXGUI.Multiplayer
+{
+    public class MapTextureCacheManager : IDisposable
+    {
+        public const int MaxCacheSize = 100;
+        public const int SleepIntervalMS = 100;
+
+        private readonly ConcurrentDictionary<Map, Image> mapTextures = [];
+
+        private readonly ConcurrentDictionary<Map, byte> missedMaps = [];
+
+        private readonly CancellationTokenSource cancellationTokenSource = new();
+
+        public MapTextureCacheManager() =>
+            Task.Run(() => MapTextureLoadingService(cancellationTokenSource.Token));
```

Comment:
There's no error handling and no task tracking. If your task errors out - there won't be any map previews anymore provided and the exception will not be caught.

Reply:
Will fix it by catching and recording exceptions in the while loop